3515/2/commands/checkin.rb

```diff
@@ -0,0 +1,85 @@
+require_relative '../helper_modules/directory_maker'
+require_relative '../helper_modules/file_write'
+module CheckinCommand
+  def checkin!(*)
+    if user_authorized?
+      checkin_false?
+    else
+      respond_with :message, text: "Извините, но ваш аккаунт не авторизован\nДля авторизации введите
+      команду /start"
+    end
+  end
+
+  def user_authorized?
+    !(User[from['id']]).to_s.empty?
+  end
+
+  def checkin_false?
+    if User[from['id']].checkin == 'false'
+      photo_request_for_checkin
+    else
+      respond_with :message, text: "Ошибка. На данный момент вы уже зачекинились в лагере\n
+      Используйте команду /checkout"
+    end
+  end
+
+  def photo_request_for_checkin
+    respond_with :message, text: 'Отправь мне селфи :)'
+    save_context :check_api_photo_file_path_for_checkin
+  end
+
+  def check_api_photo_file_path_for_checkin(*)
+    if payload.include?('photo')
+      respond_with :message, text: 'Получил фото. Сохраняю...'
+      api_photo_file_true_for_checkin(payload)
+    else
+      respond_with :message, text: 'Ну я же вежливо попросил... '
+      photo_request_for_checkin
+    end
+  end
+
+  def api_photo_file_true_for_checkin(payload)
+    photo_id = payload['photo'].last['file_id']
+    uri = URI(ENV['API_URL'] + ENV['TOKEN'] +
```

Comment:
There are a few places of such a URI building. Maybe it'd be better to place them into a few private methods, that accept all required parameters and return URIs